core-primitives/extrinsics-factory/src/mock.rs

```diff
@@ -14,24 +14,40 @@
 	limitations under the License.
 
 */
-
-use crate::{error::Result, CreateExtrinsics};
-use itp_node_api::api_client::ParentchainAdditionalParams;
-use itp_types::{parentchain::GenericMortality, OpaqueCall};
+use crate::{error::Result, AdditionParamsOf, CreateExtrinsics};
+use core::marker::PhantomData;
+use itp_types::{parentchain::GenericMortality, Balance, OpaqueCall};
 use sp_runtime::OpaqueExtrinsic;
 use std::vec::Vec;
 
+pub use itp_node_api::api_client::{
+	Config, GenericExtrinsicParams, ParentchainRuntimeConfig, PlainTip,
+};
+
 /// Mock of an extrinsics factory. To be used in unit tests.
 ///
 /// Returns an empty extrinsic.
-#[derive(Default, Clone)]
-pub struct ExtrinsicsFactoryMock;
+#[derive(Clone)]
+pub struct ExtrinsicsFactoryMock<C> {
+	_phantom: PhantomData<C>,
+}
```

Comment:
Can't derive default if the concrete type doesn't implement default. This is a well-known rust bug, and we have to implement default manually.